safe_authenticator/src/app_auth.rs

```diff
@@ -218,47 +249,86 @@ fn authenticated_app(
     app: AppInfo,
     app_id: String,
     app_container: bool,
-    _app_permissions: AppPermissions,
+    app_permissions: AppPermissions,
 ) -> Box<AuthFuture<AuthGranted>> {
     let c2 = client.clone();
     let c3 = client.clone();
+    let c4 = client.clone();
+    let client_safe_key = client.full_id();
 
     let app_keys = app.keys.clone();
     let app_pk = app.keys.public_key();
     let bootstrap_config = fry!(client::bootstrap_config());
 
     // TODO: Here, we should regenerate a token based upon reqs...
```

Comment:
With storage, this should just be 'get token from storage'.  And we shouldn't be needing to store the perms themselves at all.